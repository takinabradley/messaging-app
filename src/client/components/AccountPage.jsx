import redirectIfLoggedOut from "../modules/react-router/loaders/redirectIfLoggedOut.js"

export async function loader({ request, params }) {
  return await redirectIfLoggedOut({ request, params })
}

export default function AccountPage() {
  return (
    <div className="AccountPage" data-testid="AccountPage">
      Account stuff...
    </div>
  )
}
