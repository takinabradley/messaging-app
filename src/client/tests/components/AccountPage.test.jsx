import { describe, expect, it, test } from "vitest";
import AccountPage from "../../components/AccountPage";
import { render, screen } from "@testing-library/react";
import { createMemoryRouter, RouterProvider } from "react-router-dom"
import userEvent from "@testing-library/user-event";

describe("AccountPage", () => {
  test("renders", async () => {
    render(<AccountPage />)

    expect(screen.getByTestId("AccountPage")).toBeInTheDocument()    
    expect(screen.getByTestId("AccountPage")).toMatchSnapshot()
  })
})