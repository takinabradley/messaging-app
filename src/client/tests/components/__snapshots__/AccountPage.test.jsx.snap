// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AccountPage > renders 1`] = `
<div
  class="AccountPage"
  data-testid="AccountPage"
>
  Account stuff...
</div>
`;
